sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1750,11 +1750,10 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def invalidStreamingOutputModeError(
```

Comment:
I think this exception isn't generalized enough yet.

This now does not happen at all (since _LEGACY_ERROR_TEMP_3261 has to happen first) so it wasn't an issue, but if we deduce the meaningful error class from this, this should probably need to couple with operator, and provide the operator information as well, something along the line with this `"<operator Name> does not support <outputMode> output mode."`

Here, `<operator name>` has to be user friendly, e.g. "Streaming aggregation", not StateStoreSaveExec or so. Same for "Streaming session window aggregation", etc. 

After that, it could be bound to INVALID_STREAMING_OUTPUT_MODE.UNSUPPORTED_OPERATOR. Conceptually it has to be flipped, but if we really want to group these error classes into INVALID_STREAMING_OUTPUT_MODE. If we flip, STREAMING_OPERATOR_UNSUPPORTED_OUTPUT_MODE.